### DISCUSSION TITLE
版本号升级规则是不是有问题

### DISCUSSION THREAD

**Discussion by cuixg** [apache/doris#17026] [RFC]
1.1.5、1.2.1、1.2.2
这三个release note中，分别都包含了behaviour changes，尤其是1.1.5关于group by / having中column name和alias的变更，是一个不会报错但是影响数据产出结果的变更，在我的理解里这种变更是一个完全不向下兼容的，要么放到1.2里面才发出来，要么直接给1.1.5改成1.2的tag。因为这个升级很多sql执行结果都错掉了。大家怎么看这个问题

**Comment by morningman** [comment]
This is a good suggestion!
We should add a switch for these kind of behavior changes, and we are also aware of this, so we add a switch `group_by_and_having_use_alias_first` in 1.2.2 to try fixing this issue.

The other behavior changes are for performance and stability issue, which is reasonable for me, personally.

All release notes will be published before release for discussion and review, on Github issues.
You can also [subscribe](https://doris.apache.org/community/subscribe-mail-list) the dev@doris to get the latest news of Doris
at first time.
